### PR TITLE
LISAMZA-31528: Add top level module for improving Beam auto ELR stability and productivity

### PR DESCRIPTION
**Pull request by alnzng**
This PR is for improving the stability and productivity of current Beam auto ELR jobs at Linkedin. 

#### Problem
We have setup one auto ELR job for each module(required at Linkedin, now 15 jobs). These jobs are executed sequentially and without order guarantee. For each module, it has one dedicated Gradle task to upload it to Linkedin maven repo with no order.

For the case some ELR job is running but some modules are not uploaded yet, the ELR process will be failed. We have see this problem happened frequently which did impact the dev productivity. For example, below are some example failed ELR jobs:

- https://dependency-portal.tools.corp.linkedin.com/#/elr/details/124428
- https://dependency-portal.tools.corp.linkedin.com/#/elr/details/124429
- https://dependency-portal.tools.corp.linkedin.com/#/elr/details/124430 

#### Change
Add a top level module which includes all the required modules as dependencies, and also enforce the Gradle publish task execution order to make sure this new module always be executed lastly.

After this change, eventually we will only have one auto ELR job and will clean up the existing 15 auto ELR jobs. I would expect this change can help save 1-2 hours for each new version release.

#### Test
- https://github.com/linkedin/beam/actions/runs/5957547228/job/16160390356

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: auto-elr/build.gradle [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  implementation project(":runners:portability:java")
+
+  implementation project(":sdks:java:core")
+  implementation project(":sdks:java:extensions:join-library")
+  implementation project(":sdks:java:extensions:sorter")
+  implementation project(":sdks:java:extensions:sql")
+  implementation project(":sdks:java:io:hadoop-common")
+  implementation project(":sdks:java:io:hadoop-file-system")
+  implementation project(":sdks:java:io:hadoop-format")
+  implementation project(":sdks:java:io:kafka")
```
Comment (dxichen) [inline auto-elr/build.gradle:47]: Please also add 
`beam-runners-flink-1.15`
Context: https://jira01.corp.linkedin.com:8443/browse/TOOLS-323048

**Review by dxichen — CHANGES_REQUESTED** [review]
LGTM

I saw https://jira01.corp.linkedin.com:8443/browse/TOOLS-337460
Lets create a follow up to remove the other auto ELRs we requested previously after this is merged

File: auto-elr/build.gradle (hunk 1 above)
Comment (alnzng) [reply auto-elr/build.gradle:47]: I don't this one had been enabled in the auto ELR yet: https://engx.corp.linkedin.com/elr?search=beam-runners-flink

But we can add it and also add beam-runners-flink-1.16 there.

**Comment by alnzng** [comment]
> 



> LGTM
> 
> I saw https://jira01.corp.linkedin.com:8443/browse/TOOLS-337460 Lets create a follow up to remove the other auto ELRs we requested previously after this is merged

The job was already setup: https://engx.corp.linkedin.com/elr?search=beam-auto-elr . 
BTW, They provide a self service way to create auto ELR job, there is one checkbox you can click: https://dependency-portal.tools.corp.linkedin.com/#/elr/java

File: auto-elr/build.gradle (hunk 1 above)
Comment (alnzng) [reply auto-elr/build.gradle:47]: Added the Flink related modules.

**Comment by dxichen** [comment]
> > LGTM
> > I saw https://jira01.corp.linkedin.com:8443/browse/TOOLS-337460 Lets create a follow up to remove the other auto ELRs we requested previously after this is merged
> 
> The job was already setup: https://engx.corp.linkedin.com/elr?search=beam-auto-elr . BTW, They provide a self service way to create auto ELR job, there is one checkbox you can click: https://dependency-portal.tools.corp.linkedin.com/#/elr/java

Right I saw it was added, we need to remove the now unecessary auto elrs since only this module is required right?

**Review by dxichen — APPROVED** [review]

**Comment by alnzng** [comment]
> > > LGTM
> > > I saw https://jira01.corp.linkedin.com:8443/browse/TOOLS-337460 Lets create a follow up to remove the other auto ELRs we requested previously after this is merged
> > 
> > 
> > The job was already setup: https://engx.corp.linkedin.com/elr?search=beam-auto-elr . BTW, They provide a self service way to create auto ELR job, there is one checkbox you can click: https://dependency-portal.tools.corp.linkedin.com/#/elr/java
> 
> Right I saw it was added, we need to remove the now unecessary auto elrs since only this module is required right?

Yep, next step we need to do the clean up.

**Pull request merged**